### ISSUE TITLE
[REVIEW]: What and How of Machine Learning Transparency: Building Bespoke Explainability Tools with Interoperable Algorithmic Components

### ISSUE THREAD

**Issue by whedon** [openjournals/jose-reviews#175]
**Submitting author:** @So-Cool (<a href="http://orcid.org/0000-0002-9869-5896">Kacper Sokol</a>)
**Repository:** <a href="https://github.com/fat-forensics/Surrogates-Tutorial" target ="_blank">https://github.com/fat-forensics/Surrogates-Tutorial</a>
**Version:** 2020-ecml-pkdd
**Editor:** @ttimbers
**Reviewer:** @kvarada, @arokem
**Archive:** Pending

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://jose.theoj.org/papers/d58625bd4c600da866522c879986b18f/status.svg)](https://jose.theoj.org/papers/d58625bd4c600da866522c879986b18f)

Status badge code:

```
HTML: <a href="https://jose.theoj.org/papers/d58625bd4c600da866522c879986b18f"><img src="https://jose.theoj.org/papers/d58625bd4c600da866522c879986b18f/status.svg"></a>
Markdown: [![status](https://jose.theoj.org/papers/d58625bd4c600da866522c879986b18f/status.svg)](https://jose.theoj.org/papers/d58625bd4c600da866522c879986b18f)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/fat-forensics/Surrogates-Tutorial" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@kvarada & @arokem, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/jose-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @ttimbers know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/joss/blob/jose/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/fat-forensics/Surrogates-Tutorial">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [ ] **Version:** Does the release version given match the repository release (2020-ecml-pkdd)?
- [ ] **Authorship:** Has the submitting author (@So-Cool) made visible contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [ ] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly stated list of dependencies?
- [ ] **Usage:** Does the documentation explain how someone would adopt the module, and include examples of how to use it?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the module 2) Report issues or problems with the module 3) Seek support

### Pedagogy / Instructional design (Work-in-progress: reviewers, please comment!)

- [ ] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [ ] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [ ] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [ ] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [ ] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Does the paper clearly state the need for this module and who the target audience is?
- [ ] **Description:** Does the paper describe the learning materials and sequence?
- [ ] Does it describe how it has been used in the classroom or other settings, and how someone might adopt it?
- [ ] Could someone else teach with this module, given the right expertise?
- [ ] Does the paper tell the "story" of how the authors came to develop it, or what their expertise is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/joss/blob/jose/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source for this learning module available at the <a target="_blank" href="https://github.com/fat-forensics/Surrogates-Tutorial">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [ ] **Version:** Does the release version given match the repository release (2020-ecml-pkdd)?
- [ ] **Authorship:** Has the submitting author (@So-Cool) made visible contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation

- [ ] **A statement of need:** Do the authors clearly state the need for this module and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly stated list of dependencies?
- [ ] **Usage:** Does the documentation explain how someone would adopt the module, and include examples of how to use it?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the module 2) Report issues or problems with the module 3) Seek support

### Pedagogy / Instructional design (Work-in-progress: reviewers, please comment!)

- [ ] **Learning objectives:** Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [ ] **Content scope and length:** Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [ ] **Pedagogy:** Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [ ] **Content quality:** Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [ ] **Instructional design:** Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Does the paper clearly state the need for this module and who the target audience is?
- [ ] **Description:** Does the paper describe the learning materials and sequence?
- [ ] Does it describe how it has been used in the classroom or other settings, and how someone might adopt it?
- [ ] Could someone else teach with this module, given the right expertise?
- [ ] Does the paper tell the "story" of how the authors came to develop it, or what their expertise is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @kvarada, @arokem it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/jose-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/jose-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1145/2939672.2939778 is OK
- 10.48550/arXiv.1710.09412 is OK
- 10.48550/arXiv.1910.13016 is OK
- 10.48550/arXiv.1909.05167 is OK
- 10.21105/joss.01904 is OK
- 10.48550/arXiv.2005.01427 is OK
- 10.48550/arXiv.2008.07007 is OK
- 10.48550/arXiv.2112.14466 is OK
- 10.1023/A:1010933404324 is OK
- 10.1080/10618600.2014.907095 is OK
- 10.1214/aos/1013203451 is OK
- 10.48550/arXiv.2107.06639 is OK
- 10.1038/s42256-019-0048-x is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.07 s (193.2 files/s, 241591.3 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Jupyter Notebook                 5              0          13253           2550
Markdown                         5             98              0            690
Python                           1            145            344            242
TeX                              1             12              0            131
YAML                             2              0              0             40
-------------------------------------------------------------------------------
SUM:                            14            255          13597           3653
-------------------------------------------------------------------------------


Statistical information for the repository '2cddc32f6fb33f3953c7f434' was
gathered on 2022/06/20.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Kacper Sokol                     1           731              0          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Kacper Sokol                731          100.0          0.0               10.81
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00175/jose.00175/10.21105.jose.00175.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00175/jose.00175/10.21105.jose.00175.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
:wave: @arokem, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @kvarada, please update us on how your review is going (this is an automated reminder).

**Comment by arokem** [comment]
Just getting to it now, so you might expect my review in a couple of weeks.

**Comment by So-Cool** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00175/jose.00175/10.21105.jose.00175.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00175/jose.00175/10.21105.jose.00175.pdf) :page_facing_up: :point_left:

**Comment by So-Cool** [comment]
@arokem, the new version of the paper addresses both of your comments. [CONTRIBUTING.md](https://github.com/fat-forensics/Surrogates-Tutorial/blob/master/CONTRIBUTING.md) was added to the repository, and the *Statement of Need* section of the paper was expanded to include the missing information.

**Comment by kvarada** [comment]
@ttimbers I can't tick the check-boxes above probably because I accepted the invitation only yesterday. I'm not sure what's the best solution here. I guess I can just copy paste the items above and tick them off. Does that work for you?

**Comment by kvarada** [comment]
@So-Cool Thank you for putting together this excellent resource on this important topic. I enjoyed going through this material. 

Here are some of my initial comments: 
- The content is well motivated, well structured and easy to follow. It includes many relatable and effective examples. I like that the material has a good balance of theory and application. 
- I like that you put emphasize on the fact that the available tools need to be tuned for the given context and scenario. It might be useful to clearly show what can go wrong with one-size-fits all approach. For example, what can go wrong if I just go with SHAP explanations without carefully considering, say, interpretable representations.  
- I think the material would benefit from including more non-toy examples. 
- In general, it would be helpful for the reader if the README has more information, for example, on the license or installation. 
- Do you have the same license for the code and the teaching material? 
- I couldn't find paper.md file in the GitHub repository.

And here I'm copying the checklists, as I'm not able to tick them off above. 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSE conflict of interest policy](https://github.com/openjournals/joss/blob/jose/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSE code of conduct](https://jose.theoj.org/about#code_of_conduct).

### General checks

- [x] Repository: Is the source for this learning module available at the [repository url](https://github.com/fat-forensics/Surrogates-Tutorial)?
- [x] License: Does the repository contain a plain-text LICENSE file with the contents of a standard license? (OSI-approved for code, Creative Commons for content)
- [x] Version: Does the release version given match the repository release (2020-ecml-pkdd)?
- [x] Authorship: Has the submitting author (@So-Cool) made visible contributions to the module? Does the full list of authors seem appropriate and complete?

### Documentation
- [x] A statement of need: Do the authors clearly state the need for this module and who the target audience is?
- [x] Installation instructions: Is there a clearly stated list of dependencies?
- [x] Usage: Does the documentation explain how someone would adopt the module, and include examples of how to use it?

- [x] Community guidelines: Are there clear guidelines for third parties wishing to 1) Contribute to the module 2) Report issues or problems with the module 3) Seek support

### Pedagogy / Instructional design (Work-in-progress: reviewers, please comment!)

- [x] Learning objectives: Does the module make the learning objectives plainly clear? (We don't require explicitly written learning objectives; only that they be evident from content and design.)
- [x] Content scope and length: Is the content substantial for learning a given topic? Is the length of the module appropriate?
- [x] Pedagogy: Does the module seem easy to follow? Does it observe guidance on cognitive load? (working memory limits of 7 +/- 2 chunks of information)
- [x] Content quality: Is the writing of good quality, concise, engaging? Are the code components well crafted? Does the module seem complete?
- [x] Instructional design: Is the instructional design deliberate and apparent? For example, exploit worked-example effects; effective multi-media use; low extraneous cognitive load.

### JOSE paper

- [ ] Authors: Does the paper.md file include a list of authors with their affiliations?
- [x] A statement of need: Does the paper clearly state the need for this module and who the target audience is?
- [x] Description: Does the paper describe the learning materials and sequence?
- [x] Does it describe how it has been used in the classroom or other settings, and how someone might adopt it?
- [x] Could someone else teach with this module, given the right expertise?
- [x] Does the paper tell the "story" of how the authors came to develop it, or what their expertise is?
- [x] References: Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by So-Cool** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00175/jose.00175/10.21105.jose.00175.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00175/jose.00175/10.21105.jose.00175.pdf) :page_facing_up: :point_left:

**Comment by So-Cool** [comment]
Thanks so much for your comments, @kvarada.

- The paper is here: [`paper/paper.md`](https://github.com/fat-forensics/Surrogates-Tutorial/blob/master/paper/paper.md).
- Good catch regarding the licences. I have now fixed it and placed two badges in the README file and all of the Jupyter Notebooks.
- I have also supplemented the README file with additional details.
- The materials are focused on a toy example to facilitate an intuitive explanation of the relevant concepts without polluting the story with domain- and data-specific issues; an exploration of a real-life data set is left to the tutees as an exercise. Since each data set is unique there is no real advantage of discussing any particular scenario in detail. Nonetheless, the hope is that any specific exploration based on this collection of educational resources would link back to this repository, therefore building up a directory of problem-specific tutorials on composing bespoke explainers.
- I've added a few pointers to other (educational) materials that show how generic (one-size-fits-all) explainers may produce subpar explanations at the end of the last Jupyter Notebook (`4-explanation-generation.ipynb`).

**Comment by So-Cool** [comment]
To clarify the learning outcomes, these are now listed in the *Statement of Need* section.

@whedon generate pdf

**Comment by So-Cool** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00175/jose.00175/10.21105.jose.00175.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00175/jose.00175/10.21105.jose.00175.pdf) :page_facing_up: :point_left:

**Comment by arokem** [comment]
I have now finally had a chance to read the paper thoroughly and to go through all of the materials. I have checked off all of the boxes in my check-list and I am happy to support publication of this paper.

**Comment by labarba** [comment]
@kvarada – you have one unchecked item in [your list](https://github.com/openjournals/jose-reviews/issues/175#issuecomment-1200674854)... did you have a concern here, or you missed it?

Are you satisfied with the author's responses to your comments, at this point? Many thanks!

**Comment by kvarada** [comment]
@labarba Sorry. I just checked it. Yes O'm satisfied with the author's responses to my comments. Thanks for creating this excellent resource!

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/jose-papers/jose.00175/jose.00175/10.21105.jose.00175.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/jose-papers/blob/jose.00175/jose.00175/10.21105.jose.00175.pdf) :page_facing_up: :point_left:

**Comment by So-Cool** [comment]
The paper has been updated to include up-to-date references and acknowledgements.

@labarba, what are the next steps?

**Comment by So-Cool** [comment]
Hi @ttimbers, is there anything I can do to expedite the review/publication process?

**Comment by ttimbers** [comment]
@So-Cool - we have two things left for you to do (or perhaps you have done them and you just need to tell me about them):

1. Make a tagged release on the target repository, and report the version number here.
2. Make an archive of the repository on Zenodo or an equivalent service, and report the archive DOI here.

**Comment by So-Cool** [comment]
Hi @ttimbers, here are the details you requested:

1. GitHub release tag is `2022-jose`;
2. Zenodo DOI is `10.5281/zenodo.7229153`.

**Comment by ttimbers** [comment]
Thanks @So-Cool. I think the version needs to be done using semantic versioning however...  So something like v1.x.x - do you think you can update the version to something like that?

**Comment by So-Cool** [comment]
No worries, @ttimbers. I've updated the releases to use semantic versioning. The latest release is tagged with `1.0`.

**Comment by ttimbers** [comment]
@whedon set v1.0 as version

**Comment by whedon** [comment]
OK. v1.0 is the version.

**Comment by ttimbers** [comment]
@whedon set 10.5281/zenodo.7229153 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.7229153" target="_blank">10.5281/zenodo.7229153</a> is the archive.

**Comment by ttimbers** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1145/2939672.2939778 is OK
- 10.48550/arXiv.1710.09412 is OK
- 10.48550/arXiv.1910.13016 is OK
- 10.1016/j.simpa.2022.100406 is OK
- 10.21105/joss.01904 is OK
- 10.48550/arXiv.2005.01427 is OK
- 10.48550/arXiv.2008.07007 is OK
- 10.48550/arXiv.2112.14466 is OK
- 10.1023/A:1010933404324 is OK
- 10.1080/10618600.2014.907095 is OK
- 10.1214/aos/1013203451 is OK
- 10.48550/arXiv.2107.06639 is OK
- 10.1038/s42256-019-0048-x is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/jose-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/jose-papers/pull/106

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/jose-papers/pull/106, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by ttimbers** [comment]
@openjournals/jose-eics - I think this paper should now be ready for full acceptance. Please let me know if I missed a step with whedon.

**Comment by So-Cool** [comment]
Hi @ttimbers, is there something specific that we are waiting for before we can proceed to publication?

**Comment by labarba** [comment]
Hi @So-Cool — Apologies for the delay in executing the last step for publication. That is on me. I have read your proof one last time, and will proceed now.

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSE!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/jose-papers/pull/108
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/jose.00175](https://doi.org/10.21105/jose.00175)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by labarba** [comment]
Congratulations, @So-Cool, your JOSE paper is now published! 🚀 

Huge thanks to the Editor: @ttimbers and Reviewers: @kvarada, @arokem — your contributions are invaluable 🙏

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://jose.theoj.org/papers/10.21105/jose.00175/status.svg)](https://doi.org/10.21105/jose.00175)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/jose.00175">
  <img src="https://jose.theoj.org/papers/10.21105/jose.00175/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://jose.theoj.org/papers/10.21105/jose.00175/status.svg
   :target: https://doi.org/10.21105/jose.00175
```
This is how it will look in your documentation:

[![DOI](https://jose.theoj.org/papers/10.21105/jose.00175/status.svg)](https://doi.org/10.21105/jose.00175)

**We need your help!**

Journal of Open Source Education is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. 
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-jose

**Comment by labarba** [comment]
:loudspeaker:  New Reviewer Sign-up Form!
https://forms.gle/Rizd3TcHnQKhrbYY7
